### PR TITLE
Increase timeout for creating a new index

### LINKED ISSUE #96693: [CI] RareClusterStateIT testDeleteCreateInOneBulk failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/3e2kanyfmlmli/tests/:server:internalClusterTest/org.elasticsearch.cluster.coordination.RareClusterStateIT/testDeleteCreateInOneBulk

**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.coordination.RareClusterStateIT.testDeleteCreateInOneBulk" -Dtests.seed=E5698B5D60A58AB7 -Dtests.locale=hr-HR -Dtests.timezone=America/Edmonton -Druntime.java=17 -Dtests.fips.enabled=true
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.cluster.coordination.RareClusterStateIT&tests.test=testDeleteCreateInOneBulk

**Failure excerpt:**
```
java.util.concurrent.TimeoutException: Timeout waiting for task.

  at __randomizedtesting.SeedInfo.seed([E5698B5D60A58AB7:715645E7D63987D1]:0)
  at org.elasticsearch.action.support.PlainActionFuture$Sync.get(PlainActionFuture.java:311)
  at org.elasticsearch.action.support.PlainActionFuture.get(PlainActionFuture.java:73)
  at org.elasticsearch.cluster.coordination.RareClusterStateIT.testDeleteCreateInOneBulk(RareClusterStateIT.java:208)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by arteam**
`executeAndCancelCommittedPublication` uses 30s timeout for publication task, it makes also to use 30s timeout for request task

Fixes #96693

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ public void testDeleteCreateInOneBulk() throws Exception {
                     .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
                     .put(IndexMetadata.SETTING_WAIT_FOR_ACTIVE_SHARDS.getKey(), "0")
             ).setTimeout("0s")
-        ).get(10, TimeUnit.SECONDS);
+        ).get(30, TimeUnit.SECONDS);
```
Comment (kingherc) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java:204]: I also see several other cases of `executeAndCancelCommittedPublication()` in this test that ultimately use `.get()` with 10 seconds. Would it be reasonable to change them as well?

Comment (kingherc) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java:204]: Also, any hint as to why it takes so long >10 seconds sometimes?

Comment (kingherc) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java:204]: To clarify, I meant if you could search for the usages of `executeAndCancelCommittedPublication` and go through the subsequent .get()s to make them use 30 seconds as well. I saw like around 5 instances.

**Review by kingherc — APPROVED** [review]
LGTM

**Comment by arteam** [comment]
Thank you @kingherc !

**Pull request merged**